### ISSUE TITLE
Multiple tags on one resource

### ISSUE THREAD

**Issue by jerome-jutteau** [outscale/terraform-provider-outscale#17]
# (Imported Issue)

Hello,

The syntax for tags on our TF provider is a bit weird compared to AWS and does not allow for multiple to exists on a single resource.

AWS way:
```
tags {
  Name = "foobar"
}
```
Outscale:
```
tags {
  key = "Name"
  value = "foobar"
}
```
 Would it be possible to have the same syntax than AWS so we can add multiple tags to a single resource ?

Thanks

**Comment by Sryther** [comment]
I agree. This format is a bit weird, it implies we can only have one tag (despite the use of plural in the term `tags`).

To feed the thread, I would also suggest the following format:

```
tags = [
  {
    key = "Name"
    value = "foobar"
  },
  {
    key = "Name2"
    value = "foobar"
  }
]
```

But I would prefere like you said the AWS/Openstack format:

```
tags {
  name = "foobar"
  name2 = "foobar"
}
```

This behavior is blocking and limits the use of tags.

**Comment by jerome-jutteau** [comment]
For newcomers, note that it is still possible to add multiple tags in [outscale_vm](https://registry.terraform.io/providers/outscale-dev/outscale/latest/docs/resources/vm) adding multiple `tags` blocks:
```
  tags {
    key   = "osc.fcu.eip.auto-attach"
    value = outscale_public_ip.my_public_ip.public_ip
  }
  tags {
    key   = "my.key"
    value = "my.value"
  }
```

**Comment by cvedia-mdsol** [comment]
Is there any chance the tagging on a resource would be able to support specifying an array of values instead of having to add multiple blocks of "tags" similar to what Sryther mentioned? The (AWS/Openstack) format would especially keep the configuration a bit cleaner when it comes to adding a larger set of tags.

**Comment by outscale-rce** [comment]
Thanks for your suggestion. The current tagging format follows our API design, where tags are structured as a list of key-value pairs. This ensures consistency across resources and aligns with how our API processes tags.

We understand the preference for specifying an array instead of multiple blocks, but at the moment, this isn't supported. If there are any updates or changes regarding this, we’ll be sure to share them.

Let us know if you have any further questions!

Thanks!
